# Lab 4 Data Cleaning

#### October 29th, 2019

##### Ziah Daily

## Introduction

### Abstract





# Dataset Exploration

In [537]:
# always include aliased modules
# these are the most common ones that we'll use in our Big Data class
import math as m
import numpy as np
import scipy as sp
import pandas as pd
import matplotlib as mpl
import matplotlib.pyplot as plt

In [538]:
#In this part of the code I am opening all three datasets in order to get an idea of what it looks like.

attendance_df = pd.read_excel("attendance (1).xls") 

sales_df = pd.read_csv("sales.csv", encoding = "iso-8859-1")

boing_df = pd.read_excel("BOING-BOING-CANDY-HIERARCHY-2016-SURVEY-Responses.xlsx", encoding = "iso-8859-1")

In the data below I wanted to look at the first seven rows after the header in order to figure out what the attendance for each of the datasets. Additionally, I used the info function on all three datasets in order to find out additonal information.

In [539]:
#I am looking at the first 9 rows of data for the attenednce data
attendance_df.head(9)

Table 43. Average daily attendance (ADA) as a percentage of total enrollment, school day length, and school year length in public schools, by school level and state: 2007-08  \
0                                                NaN                                                                                                                              
1                                                NaN                                                                                                                              
2                                                  1                                                                                                                              
3                             United States ........                                                                                                                              
4                          Alabama .................                                                                                                                              
5                          Alaska ..................                                                                                                                              
6                          Arizona .................                                                                                                                              
7                          Arkansas ................                                                                                                                              
8                          California ..............                                                                                                                              

                                          Unnamed: 1  Unnamed: 2  \
0  Total elementary, secondary, and combined elem...         NaN   
1                       ADA as percent of enrollment         NaN   
2                                                  2         NaN   
3                                             93.079    0.218555   
4                                            93.8124    1.237261   
5                                            89.9176    1.215976   
6                                             89.037    2.952601   
7                                            91.8271    1.352924   
8                                             93.241    0.711649   

                    Unnamed: 3  Unnamed: 4                   Unnamed: 5  \
0                          NaN         NaN                          NaN   
1  Average hours in school day         NaN  Average days in school year   
2                            3         NaN                            4   
3                       6.6447    0.017650                          180   
4                      7.02852    0.065621                          180   
5                      6.47688    0.049882                          180   
6                      6.43369    0.091917                          181   
7                      6.88542    0.061468                          179   
8                      6.24064    0.066364                          181   

   Unnamed: 6                    Unnamed: 7  Unnamed: 8  \
0         NaN                           NaN         NaN   
1         NaN  Average hours in school year         NaN   
2         NaN                             5         NaN   
3    0.142993                       1192.65      3.0859   
4    0.755110                       1266.62     12.3343   
5    3.427735                       1162.91     22.8639   
6    1.678664                       1159.11     14.4327   
7    0.201577                       1228.89     10.6821   
8    0.439441                       1128.77     12.5276   

                     Unnamed: 9  Unnamed: 10                  Unnamed: 11  \
0            Elementary schools          NaN                          NaN   
1  ADA as percent of enrollment   

After looking at the first 7 rows of the attendance data, it is difficult to see what the dataset is showing, but looking below at other rows, you can make an educated guess.

In [540]:
#This piece of code looks at addiotnal information that might helpful (i.e title of dataset and how many entries there are)
attendance_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 68 entries, 0 to 67
Data columns (total 17 columns):
Table 43. Average daily attendance (ADA) as a percentage of total enrollment, school day length, and school year length in public schools, by school level and state: 2007-08    57 non-null object
Unnamed: 1                                                                                                                                                                       55 non-null object
Unnamed: 2                                                                                                                                                                       52 non-null float64
Unnamed: 3                                                                                                                                                                       54 non-null object
Unnamed: 4                                                                                                      

In [541]:
#Similar to the attendance dataset above, the code looks at the first 6 lines of data in the sales dataset.
sales_df.head(6)

Unnamed: 0              event_id        primary_act_id  \
0           1  abcaf1adb99a935fc661  43f0436b905bfa7c2eec   
1           2  6c56d7f08c95f2aa453c  1a3e9aecd0617706a794   
2           3  c7ab4524a121f9d687d2  4b677c3f5bec71eec8d1   
3           4  394cb493f893be9b9ed1  b1ccea01ad6ef8522796   
4           5  55b5f67e618557929f48  91c03a34b562436efa3c   
5           6  4f10fd8b9f550352bd56  ac4b847b3fde66f2117e   

       secondary_act_id   purch_party_lkup_id  \
0  b85143bf51323b72e53c  7dfa56dd7d5956b17587   
1  f53529c5679ea6ca5a48  4f9e6fc637eaf7b736c2   
2  b85143bf51323b72e53c  6c2545703bd527a7144d   
3  b85143bf51323b72e53c  527d6b1eaffc69ddd882   
4  b85143bf51323b72e53c  8bd62c394a35213bdf52   
5  63814f3d63317f1b56c4  3b3a628f83135acd0676   

                                          event_name  \
0  Xfinity Center Mansfield Premier Parking: Flor...   
1      Gorge Camping - dave matthews band - sept 3-7   
2        Dodge Theatre Adams Street Parking - benise   
3  Gexa Energy Pavilion Vip Parking : kid rock wi...   
4                      Premier Parking - motley crue   
5                          Fast Lane Access: Journey   

                           primary_act_name  secondary_act_name  \
0  XFINITY Center Mansfield Premier Parking                 NaN   
1                             Gorge Camping  Dave Matthews Band   
2                             Parking Event                 NaN   
3          Gexa Energy Pavilion VIP Parking                 NaN   
4  White River Amphitheatre Premier Parking                 NaN   
5                          Fast Lane Access             Journey   

  major_cat_name          minor_cat_name     ...     edu_1st_indv_val  \
0           MISC                 PARKING     ...                  NaN   
1           MISC                 CAMPING     ...                  NaN   
2           MISC                 PARKING     ...                  NaN   
3           MISC                 PARKING     ...                  NaN   
4           MISC                 PARKING     ...                  NaN   
5           MISC  SPECIAL ENTRY (UPSELL)     ...                  NaN   

  edu_2nd_indv_val adults_in_hh_num  married_ind  child_present_ind  \
0              NaN              NaN          NaN                NaN   
1              NaN              NaN          NaN                NaN   
2              NaN              NaN          NaN                NaN   
3              NaN              NaN          NaN                NaN   
4              NaN              NaN          NaN                NaN   
5              NaN              NaN          NaN                NaN   

  home_owner_ind occpn_val occpn_1st_val occpn_2nd_val dist_to_ven  
0            NaN       NaN           NaN           NaN         NaN  
1            NaN       NaN           NaN           NaN        59.0  
2            NaN       NaN           NaN           NaN         NaN  
3            NaN       NaN           NaN           NaN         NaN  
4            NaN       NaN           NaN           NaN         NaN  
5            NaN       NaN           NaN           NaN         NaN  

[6 rows x 46 columns]

In [542]:
#To go into more depth on the sales data I look at info, which tells me how many entries there are. 
sales_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 46 columns):
Unnamed: 0                5000 non-null int64
event_id                  5000 non-null object
primary_act_id            5000 non-null object
secondary_act_id          5000 non-null object
purch_party_lkup_id       5000 non-null object
event_name                5000 non-null object
primary_act_name          5000 non-null object
secondary_act_name        1586 non-null object
major_cat_name            5000 non-null object
minor_cat_name            5000 non-null object
la_event_type_cat         5000 non-null object
event_disp_name           5000 non-null object
ticket_text               5000 non-null object
tickets_purchased_qty     5000 non-null int64
trans_face_val_amt        5000 non-null float64
delivery_type_cd          5000 non-null object
event_date_time           5000 non-null object
event_dt                  5000 non-null object
presale_dt                2108 non-null object
o

Looking at the sales dataset it looks like there is a lot of dataset  missing, especially in the last 14 collumns of data. Additonally it looks like there is a lot of informaion in this dataset, 5000 rows.

In [543]:
#next I am looking at the first 6 rows of the BOING BOING CANDY dataset.
boing_df.head(6)

Timestamp  \
0 2016-10-24 05:09:23.033   
1 2016-10-24 05:09:54.798   
2 2016-10-24 05:13:06.734   
3 2016-10-24 05:14:17.192   
4 2016-10-24 05:14:24.625   
5 2016-10-24 05:16:43.126   

  Are you going actually going trick or treating yourself? Your gender:  \
0                                                 No               Male   
1                                                 No               Male   
2                                                 No             Female   
3                                                 No               Male   
4                                                Yes               Male   
5                                                 No               Male   

  How old are you? Which country do you live in?  \
0               22                        Canada   
1               45                           usa   
2               48                            US   
3               57                           usa   
4               42                           USA   
5               41                           USA   

  Which state, province, county do you live in?  [100 Grand Bar]  \
0                                       Ontario              JOY   
1                                            il              MEH   
2                                      Colorado              JOY   
3                                            il              JOY   
4                                  South Dakota              MEH   
5                                       Georgia              JOY   

   [Anonymous brown globs that come in black and orange wrappers]  \
0                                            DESPAIR                
1                                                MEH                
2                                            DESPAIR                
3                                                MEH                
4                                            DESPAIR                
5                                            DESPAIR                

   [Any full-sized candy bar]  [Black Jacks]  \
0                         JOY            MEH   
1                         JOY            JOY   
2                         JOY            MEH   
3                         JOY            MEH   
4                         JOY        DESPAIR   
5                         JOY            MEH   

                 ...                 \
0                ...                  
1                ...                  
2                ...                  
3                ...                  
4                ...                  
5                ...                  

  Please estimate the degree(s) of separation you have from the following celebrities [JK Rowling]  \
0                                        3 or higher                                                 
1                                        3 or higher                                                 
2                                        3 or higher                                                 
3                                        3 or higher                                                 
4                                        3 or higher                                                 
5                                        3 or higher                                                 

  Please estimate the degree(s) of separation you have from the following celebrities [JJ Abrams]  \
0                                                  2                                                
1                                        3 or higher                                                
2                                        3 or higher                                                
3                                        3 or higher                                                
4                                        3 or higher                                                
5                           

In [544]:
#This peice of code tells me information such as amount of entries in the dataset and number of columns.
boing_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1259 entries, 0 to 1258
Columns: 123 entries, Timestamp to  [York Peppermint Patties] Ignore
dtypes: datetime64[ns](1), float64(1), object(121)
memory usage: 1.2+ MB


Looking at the info function for BOING BOING CANDy dataset, it doesn't give me a lot of information. Given that the previous two datasets gave me more information, I probably will not cean up this one and instead choose the last two datassets (to clean up).

# Dataset Preperation 1

One of the two datasets that I chose to use is attendance. For this dataset I have decided to rename all of the columns.

In [545]:
# this code deletes the first four rows fo code by using drop.

attendance_df = attendance_df.drop(attendance_df.index[0:3])

# afterwards I want to see if the code, worked so I print the data
attendance_df.head(6)

Table 43. Average daily attendance (ADA) as a percentage of total enrollment, school day length, and school year length in public schools, by school level and state: 2007-08  \
3                             United States ........                                                                                                                              
4                          Alabama .................                                                                                                                              
5                          Alaska ..................                                                                                                                              
6                          Arizona .................                                                                                                                              
7                          Arkansas ................                                                                                                                              
8                          California ..............                                                                                                                              

  Unnamed: 1  Unnamed: 2 Unnamed: 3  Unnamed: 4 Unnamed: 5  Unnamed: 6  \
3     93.079    0.218555     6.6447    0.017650        180    0.142993   
4    93.8124    1.237261    7.02852    0.065621        180    0.755110   
5    89.9176    1.215976    6.47688    0.049882        180    3.427735   
6     89.037    2.952601    6.43369    0.091917        181    1.678664   
7    91.8271    1.352924    6.88542    0.061468        179    0.201577   
8     93.241    0.711649    6.24064    0.066364        181    0.439441   

  Unnamed: 7  Unnamed: 8 Unnamed: 9  Unnamed: 10 Unnamed: 11  Unnamed: 12  \
3    1192.65      3.0859     94.005     0.269369     6.65606     0.015998   
4    1266.62     12.3343    93.7764     1.839154     7.03847     0.075944   
5    1162.91     22.8639    91.2815     1.563482     6.48679     0.053082   
6    1159.11     14.4327    88.9044     3.905033     6.44315     0.102053   
7    1228.89     10.6821    92.0932     2.091651     6.93216     0.076546   
8    1128.77     12.5276    94.9314     0.753808     6.28974     0.051471   

  Unnamed: 13 Unnamed: 14 Unnamed: 15 Unnamed: 16  
3     91.1181    0.432223     6.59439    0.040308  
4      94.561     0.37928     7.13724     0.17312  
5     93.2409     1.57371     6.24087    0.145993  
6     88.9797     3.22058     6.36909    0.248824  
7      90.812     2.23155     6.76751    0.095928  
8     89.3609     1.44673     6.14262    0.202711

In [546]:
# new names for the columns/headers

attendance_df.rename(columns ={
    attendance_df.columns[0]:"Place",
    attendance_df.columns[1]:"ADA as percent of enrollment", 
    attendance_df.columns[2]:"NaN",
    attendance_df.columns[3]:"Average days in school year", 
    attendance_df.columns[4]:"NaN",
    attendance_df.columns[5]:"ADA as percent of enrollment-Elementary schools", 
    attendance_df.columns[6]:'NaN',
    attendance_df.columns[7]:"ADA as percent of enrollment-Secondary schools",
    attendance_df.columns[8]:"NaN",
    attendance_df.columns[9]:"Average hours in school day-Secondary schools",
    attendance_df.columns[10]:"NaN",
    attendance_df.columns[11]:"Average hours in school day-Secondary schools",
    attendance_df.columns[12]:"NaN",
    attendance_df.columns[13]:"Average hours in school day-Secondary schools",
    attendance_df.columns[14]:"NaN",
    attendance_df.columns[15]:"Average hours in school day-Secondary schools",
    attendance_df.columns[16]:"NaN"
    },

            
        inplace = True)

#Now I want to print the first 9 rows to make sure that the code worked.
attendance_df.head(9)

Place ADA as percent of enrollment       NaN  \
3      United States ........                       93.079  0.218555   
4   Alabama .................                      93.8124  1.237261   
5   Alaska ..................                      89.9176  1.215976   
6   Arizona .................                       89.037  2.952601   
7   Arkansas ................                      91.8271  1.352924   
8   California ..............                       93.241  0.711649   
9                         NaN                          NaN       NaN   
10  Colorado ................                      93.8872  0.440366   
11  Connecticut .............                      87.9259  2.984395   

   Average days in school year       NaN  \
3                       6.6447  0.017650   
4                      7.02852  0.065621   
5                      6.47688  0.049882   
6                      6.43369  0.091917   
7                      6.88542  0.061468   
8                      6.24064  0.066364   
9                          NaN       NaN   
10                     7.00761  0.051954   
11                     6.46655  0.087829   

   ADA as percent of enrollment-Elementary schools       NaN  \
3                                              180  0.142993   
4                                              180  0.755110   
5                                              180  3.427735   
6                                              181  1.678664   
7                                              179  0.201577   
8                                              181  0.439441   
9                                              NaN       NaN   
10                                             171  1.047398   
11                                             181  0.137733   

   ADA as percent of enrollment-Secondary schools      NaN  \
3                                         1192.65   3.0859   
4                                         1266.62  12.3343   
5                                         1162.91  22.8639   
6                                         1159.11  14.4327   
7                                         1228.89  10.6821   
8                                         1128.77  12.5276   
9                                             NaN      NaN   
10                                         1199.3   9.9247   
11                                        1173.08  15.9229   

   Average hours in school day-Secondary schools       NaN  \
3                                         94.005  0.269369   
4                                        93.7764  1.839154   
5                                        91.2815  1.563482   
6                                        88.9044  3.905033   
7                                        92.0932  2.091651   
8                                        94.9314  0.753808   
9                                            NaN       NaN   
10                                       94.5353  0.447224   
11                                       87.3722  3.977535   

   Average hours in school day-Secondary schools       NaN  \
3                                        6.65606  0.015998   
4                                        7.03847  0.075944   
5                                        6.48679  0.053082   
6                                        6.44315  0.102053   
7                                        6.93216  0.076546   
8                                        6.28974  0.051471   
9                                            NaN       NaN   
10                                       6.96058  0.066373   
11                                       6.46963  0.111509   

   Average hours in school day-Secondary schools       NaN  \
3                                        91.1181  0.432223   
4                                         94.561   0.37928   
5                                        93.2409   1.57371   
6                                        88.9797   3.22058   
7                                         90.812

In [547]:
#For the code below I want to drop some all of the collumns witht he name NaN. I do this becuase if I dont know what the data is about it isnt relevent and therefore should be dropped.
attendance_df = attendance_df.drop("NaN",axis=1)

#Then I want to print the first 9 ros to make sure that my code worked
attendance_df.head(9)

Place ADA as percent of enrollment  \
3      United States ........                       93.079   
4   Alabama .................                      93.8124   
5   Alaska ..................                      89.9176   
6   Arizona .................                       89.037   
7   Arkansas ................                      91.8271   
8   California ..............                       93.241   
9                         NaN                          NaN   
10  Colorado ................                      93.8872   
11  Connecticut .............                      87.9259   

   Average days in school year  \
3                       6.6447   
4                      7.02852   
5                      6.47688   
6                      6.43369   
7                      6.88542   
8                      6.24064   
9                          NaN   
10                     7.00761   
11                     6.46655   

   ADA as percent of enrollment-Elementary schools  \
3                                              180   
4                                              180   
5                                              180   
6                                              181   
7                                              179   
8                                              181   
9                                              NaN   
10                                             171   
11                                             181   

   ADA as percent of enrollment-Secondary schools  \
3                                         1192.65   
4                                         1266.62   
5                                         1162.91   
6                                         1159.11   
7                                         1228.89   
8                                         1128.77   
9                                             NaN   
10                                         1199.3   
11                                        1173.08   

   Average hours in school day-Secondary schools  \
3                                         94.005   
4                                        93.7764   
5                                        91.2815   
6                                        88.9044   
7                                        92.0932   
8                                        94.9314   
9                                            NaN   
10                                       94.5353   
11                                       87.3722   

   Average hours in school day-Secondary schools  \
3                                        6.65606   
4                                        7.03847   
5                                        6.48679   
6                                        6.44315   
7                                        6.93216   
8                                        6.28974   
9                                            NaN   
10                                       6.96058   
11                                       6.46963   

   Average hours in school day-Secondary schools  \
3                                        91.1181   
4                                         94.561   
5                                        93.2409   
6                                        88.9797   
7                                         90.812   
8                                        89.3609   
9                                            NaN   
10                                       91.2429   
11                                       93.7472   

   Average hours in school day-Secondary schools  
3                                        6.59439  
4                                        7.13724  
5                                        6.24087  
6                                        6.36909  
7                                        6.76751  
8                                        6.14262  
9                                            NaN  
10                 

In [548]:
# new names for the columns/headers

attendance_df.rename(columns ={
    attendance_df.columns[0]:"Place",
    attendance_df.columns[1]:"ADA as percent of enrollment", 
    attendance_df.columns[2]:"Average hours in school day",
    attendance_df.columns[3]:"Average days in school year", 
    attendance_df.columns[4]:"Average hours in school year",
    attendance_df.columns[5]:"ADA as percent of enrollment-Elementary schools", 
    attendance_df.columns[6]:'Average hours in school day-Elementary schools',
    attendance_df.columns[7]:"ADA as percent of enrollment-Secondary schools",
    attendance_df.columns[7]:"Average hours in school day-Secondary schools"
    },
            
        inplace = True)

#Now I want to print the first 9 rows of data to mak sure that thee code above works
attendance_df.head(9)

Place ADA as percent of enrollment  \
3      United States ........                       93.079   
4   Alabama .................                      93.8124   
5   Alaska ..................                      89.9176   
6   Arizona .................                       89.037   
7   Arkansas ................                      91.8271   
8   California ..............                       93.241   
9                         NaN                          NaN   
10  Colorado ................                      93.8872   
11  Connecticut .............                      87.9259   

   Average hours in school day Average days in school year  \
3                       6.6447                         180   
4                      7.02852                         180   
5                      6.47688                         180   
6                      6.43369                         181   
7                      6.88542                         179   
8                      6.24064                         181   
9                          NaN                         NaN   
10                     7.00761                         171   
11                     6.46655                         181   

   Average hours in school year Average hours in school day-Secondary schools  \
3                       1192.65                                        94.005   
4                       1266.62                                       93.7764   
5                       1162.91                                       91.2815   
6                       1159.11                                       88.9044   
7                       1228.89                                       92.0932   
8                       1128.77                                       94.9314   
9                           NaN                                           NaN   
10                       1199.3                                       94.5353   
11                      1173.08                                       87.3722   

   Average hours in school day-Secondary schools  \
3                                        6.65606   
4                                        7.03847   
5                                        6.48679   
6                                        6.44315   
7                                        6.93216   
8                                        6.28974   
9                                            NaN   
10                                       6.96058   
11                                       6.46963   

   Average hours in school day-Secondary schools  \
3                                        91.1181   
4                                         94.561   
5                                        93.2409   
6                                        88.9797   
7                                         90.812   
8                                        89.3609   
9                                            NaN   
10                                       91.2429   
11                                       93.7472   

   Average hours in school day-Secondary schools  
3                                        6.59439  
4                                        7.13724  
5                                        6.24087  
6                                        6.36909  
7                                        6.76751  
8                                        6.14262  
9                                            NaN  
10                                       7.00875  
11                                       6.46219

In [549]:
#Now I want to use the groupby function in order group by "place"
attendance_df.groupby("Average days in school year").count()

Place  ADA as percent of enrollment  \
Average days in school year                                        
171                              1                             1   
172                              1                             1   
173                              2                             2   
175                              1                             1   
176                              4                             4   
177                              4                             4   
178                              4                             4   
179                              4                             4   
180                             16                            16   
181                             11                            11   
182                              3                             3   
184                              1                             1   

                             Average hours in school day  \
Average days in school year                                
171                                                    1   
172                                                    1   
173                                                    2   
175                                                    1   
176                                                    4   
177                                                    4   
178                                                    4   
179                                                    4   
180                                                   16   
181                                                   11   
182                                                    3   
184                                                    1   

                             Average hours in school year  \
Average days in school year                                 
171                                                     1   
172                                                     1   
173                                                     2   
175                                                     1   
176                                                     4   
177                                                     4   
178                                                     4   
179                                                     4   
180                                                    16   
181                                                    11   
182                                                     3   
184                                                     1   

                             Average hours in school day-Secondary schools  \
Average days in school year                                                  
171                                                                      1   
172                                                                      1   
173                                                                      2   
175                                                                      1   
176                                                                      4   
177                                                                      4   
178                                                                      4   
179                                                                      4   
180                                                                     16   
181                                                                     11   
182                                                                      3   
184                                                                      1   

                             Average hours in school day-Secondary schools  \
Average days in school year                                                  
171                                                                      1   
172                                                  

After grouping by place I found that having 180 schooldaays is the most common for all of the states. The next highest amount of states with a certain amount of schooldays was 181 school days.

# Dataset Preperation 2

In the second dataset I want to work on the sales dataset. I chose this dataset becuase I saw some really potential to learn intrestin things about this dataset.

In [550]:
#the code below drops certain collumns that I do not need. I chose to drop these particular columns so that I can I focus on the information that I find more intrsting and uselful.
sales_df = sales_df.drop("gndr_cd",axis=1)
sales_df = sales_df.drop("age_yr",axis=1)
sales_df = sales_df.drop("income_amt",axis=1)
sales_df = sales_df.drop("edu_val",axis=1)
sales_df = sales_df.drop("edu_1st_indv_val",axis=1)
sales_df = sales_df.drop("edu_2nd_indv_val",axis=1)
sales_df = sales_df.drop("adults_in_hh_num",axis=1)
sales_df = sales_df.drop("married_ind",axis=1)
sales_df = sales_df.drop("child_present_ind",axis=1)
sales_df = sales_df.drop("home_owner_ind",axis=1)
sales_df = sales_df.drop("occpn_val",axis=1)
sales_df = sales_df.drop("occpn_1st_val",axis=1)
sales_df = sales_df.drop("occpn_2nd_val",axis=1)
sales_df = sales_df.drop("dist_to_ven",axis=1)

#Next I wan to print the first 9 rows of the code to make sure that it works.
sales_df.head(9)

Unnamed: 0              event_id        primary_act_id  \
0           1  abcaf1adb99a935fc661  43f0436b905bfa7c2eec   
1           2  6c56d7f08c95f2aa453c  1a3e9aecd0617706a794   
2           3  c7ab4524a121f9d687d2  4b677c3f5bec71eec8d1   
3           4  394cb493f893be9b9ed1  b1ccea01ad6ef8522796   
4           5  55b5f67e618557929f48  91c03a34b562436efa3c   
5           6  4f10fd8b9f550352bd56  ac4b847b3fde66f2117e   
6           7  105ea9ee10065b54b0df  a14232befff04be1e2f3   
7           8  593b910d17913d7101cb  91c03a34b562436efa3c   
8           9  0fb4e7152e09a40ffedd  0efaba7ce3f0d7466b42   

       secondary_act_id   purch_party_lkup_id  \
0  b85143bf51323b72e53c  7dfa56dd7d5956b17587   
1  f53529c5679ea6ca5a48  4f9e6fc637eaf7b736c2   
2  b85143bf51323b72e53c  6c2545703bd527a7144d   
3  b85143bf51323b72e53c  527d6b1eaffc69ddd882   
4  b85143bf51323b72e53c  8bd62c394a35213bdf52   
5  63814f3d63317f1b56c4  3b3a628f83135acd0676   
6  b85143bf51323b72e53c  14af02b72584fa823f85   
7  b85143bf51323b72e53c  c7878d51bb2d62d4705c   
8  f66a1cf484a8cd868fb4  ec216826c73d9a3f6f9c   

                                          event_name  \
0  Xfinity Center Mansfield Premier Parking: Flor...   
1      Gorge Camping - dave matthews band - sept 3-7   
2        Dodge Theatre Adams Street Parking - benise   
3  Gexa Energy Pavilion Vip Parking : kid rock wi...   
4                      Premier Parking - motley crue   
5                          Fast Lane Access: Journey   
6  Phil Lesh & friends-pnc bank arts center holmd...   
7  Star Parking Presented By Hertz: Train -  whit...   
8                Brad Paisley: Hertz Premier Parking   

                               primary_act_name  secondary_act_name  \
0      XFINITY Center Mansfield Premier Parking                 NaN   
1                                 Gorge Camping  Dave Matthews Band   
2                                 Parking Event                 NaN   
3              Gexa Energy Pavilion VIP Parking                 NaN   
4      White River Amphitheatre Premier Parking                 NaN   
5                              Fast Lane Access             Journey   
6  PNC Bank Arts Center Holmdel Premier Parking                 NaN   
7      White River Amphitheatre Premier Parking                 NaN   
8                         Hertz Premier Parking        Brad Paisley   

  major_cat_name          minor_cat_name          ...           print_dt  \
0           MISC                 PARKING          ...            9/12/15   
1           MISC                 CAMPING          ...             9/1/09   
2           MISC                 PARKING          ...             4/5/06   
3           MISC                 PARKING          ...             7/6/11   
4           MISC                 PARKING          ...            6/28/05   
5           MISC  SPECIAL ENTRY (UPSELL)          ...            7/21/12   
6           MISC                 PARKING          ...            6/16/06   
7           MISC                 PARKING          ...             3/4/13   
8           MISC                 PARKING          ...            8/20/12   

  timezn_nm      venue_city    venue_state  venue_postal_cd_sgmt_1  \
0       EST       MANSFIELD  MASSACHUSETTS                    2048   
1       PST          QUINCY     WASHINGTON                   98848   
2       MST         PHOENIX        ARIZONA                   85003   
3       CST          DALLAS          TEXAS                   75210   
4       PST          AUBURN     WASHINGTON                   98092   
5       PST  SAN BERNARDINO     CALIFORNIA                   92407   
6       EST         HOLMDEL     NEW JERSEY                    7733   
7       PST          AUBURN     WASHINGTON                   98092   
8       MST         PHOENIX        ARIZONA                   85035   

             sales_platform_cd print_flg la_valid_tkt_event_flg   fin_mkt_nm  \
0  www.concerts.livenation.com        T                      N        Boston   
1      

In [551]:
# new names for the columns/headers

sales_df.rename(columns ={
    sales_df.columns[1]:"Event", 
    sales_df.columns[2]:"Primary Act ID",
    sales_df.columns[3]:"Secondary Act ID", 
    sales_df.columns[4]:"Purch Party",
    sales_df.columns[5]:"event name",
    sales_df.columns[6]:"Name of Primary Act", 
    sales_df.columns[7]:"Name of Secondary Act",
    sales_df.columns[8]:"Major Cat Name",
    sales_df.columns[9]:"Minor Cat Name",
    sales_df.columns[11]:"Print date",
    sales_df.columns[12]:"timezone",
    sales_df.columns[13]:"date",
    sales_df.columns[14]:"venue city",
    sales_df.columns[15]:"venue state",
    sales_df.columns[16]:"Venue postal code",
    sales_df.columns[17]:"Sales platform code",
    sales_df.columns[18]:"Print Flag",
    sales_df.columns[19]:"la valid ticket event flag",
    sales_df.columns[20]:"fin maket number",
    sales_df.columns[21]:"website session cookie validation"},
   
            
        inplace = True)

#Nect I want to print the first 5 rows of this datset to see if my code worked.
sales_df.head(5)

Unnamed: 0                 Event        Primary Act ID  \
0           1  abcaf1adb99a935fc661  43f0436b905bfa7c2eec   
1           2  6c56d7f08c95f2aa453c  1a3e9aecd0617706a794   
2           3  c7ab4524a121f9d687d2  4b677c3f5bec71eec8d1   
3           4  394cb493f893be9b9ed1  b1ccea01ad6ef8522796   
4           5  55b5f67e618557929f48  91c03a34b562436efa3c   

       Secondary Act ID           Purch Party  \
0  b85143bf51323b72e53c  7dfa56dd7d5956b17587   
1  f53529c5679ea6ca5a48  4f9e6fc637eaf7b736c2   
2  b85143bf51323b72e53c  6c2545703bd527a7144d   
3  b85143bf51323b72e53c  527d6b1eaffc69ddd882   
4  b85143bf51323b72e53c  8bd62c394a35213bdf52   

                                          event name  \
0  Xfinity Center Mansfield Premier Parking: Flor...   
1      Gorge Camping - dave matthews band - sept 3-7   
2        Dodge Theatre Adams Street Parking - benise   
3  Gexa Energy Pavilion Vip Parking : kid rock wi...   
4                      Premier Parking - motley crue   

                        Name of Primary Act Name of Secondary Act  \
0  XFINITY Center Mansfield Premier Parking                   NaN   
1                             Gorge Camping    Dave Matthews Band   
2                             Parking Event                   NaN   
3          Gexa Energy Pavilion VIP Parking                   NaN   
4  White River Amphitheatre Premier Parking                   NaN   

  Major Cat Name Minor Cat Name          ...           print_dt timezn_nm  \
0           MISC        PARKING          ...            9/12/15       EST   
1           MISC        CAMPING          ...             9/1/09       PST   
2           MISC        PARKING          ...             4/5/06       MST   
3           MISC        PARKING          ...             7/6/11       CST   
4           MISC        PARKING          ...            6/28/05       PST   

  venue_city    venue_state  venue_postal_cd_sgmt_1  \
0  MANSFIELD  MASSACHUSETTS                    2048   
1     QUINCY     WASHINGTON                   98848   
2    PHOENIX        ARIZONA                   85003   
3     DALLAS          TEXAS                   75210   
4     AUBURN     WASHINGTON                   98092   

             sales_platform_cd print_flg la_valid_tkt_event_flg fin_mkt_nm  \
0  www.concerts.livenation.com        T                      N      Boston   
1                          NaN        T                      N     Seattle   
2                          NaN        T                      N     Arizona   
3                          NaN        T                      N      Dallas   
4                          NaN        T                      N     Seattle   

  web_session_cookie_val  
0   7dfa56dd7d5956b17587  
1   4f9e6fc637eaf7b736c2  
2   6c2545703bd527a7144d  
3   527d6b1eaffc69ddd882  
4   8bd62c394a35213bdf52  

[5 rows x 32 columns]

In [552]:
#Next I use the code to group all the date by venue state.
sales_df.groupby("venue_state").count()

Unnamed: 0  Event  Primary Act ID  Secondary Act ID  \
venue_state                                                                 
ALABAMA                       21     21              21                21   
ALBERTA                       29     29              29                29   
ARIZONA                      214    214             214               214   
ARKANSAS                       2      2               2                 2   
BRITISH COLUMBIA              39     39              39                39   
CALIFORNIA                   712    712             712               712   
COLORADO                      34     34              34                34   
CONNECTICUT                  144    144             144               144   
DISTRICT OF COLUMBIA          30     30              30                30   
FLORIDA                      236    236             236               236   
GEORGIA                      213    213             213               213   
ILLINOIS                     227    227             227               227   
INDIANA                      296    296             296               296   
IOWA                           1      1               1                 1   
KANSAS                        21     21              21                21   
KENTUCKY                      17     17              17                17   
LOUISIANA                     26     26              26                26   
MAINE                          7      7               7                 7   
MANITOBA                       8      8               8                 8   
MARYLAND                      32     32              32                32   
MASSACHUSETTS                229    229             229               229   
MICHIGAN                      72     72              72                72   
MINNESOTA                     14     14              14                14   
MISSISSIPPI                    2      2               2                 2   
MISSOURI                      95     95              95                95   
NEBRASKA                       5      5               5                 5   
NEVADA                        41     41              41                41   
NEW JERSEY                   195    195             195               195   
NEW MEXICO                    39     39              39                39   
NEW YORK                     381    381             381               381   
NORTH CAROLINA               293    293             293               293   
OHIO                         170    170             170               170   
OKLAHOMA                       9      9               9                 9   
ONTARIO                       77     77              77                77   
OREGON                         9      9               9                 9   
PENNSYLVANIA                 225    225             225               225   
PRINCE EDWARD ISLAND           1      1               1                 1   
QUEBEC                         5      5               5                 5   
RHODE ISLAND                   2      2               2                 2   
SASKATCHEWAN                   2      2               2                 2   
SOUTH CAROLINA                17     17              17                17   
SOUTH DAKOTA                   1      1               1                 1   
TENNESSEE                     39     39              39                39   
TEXAS                        270    270             270               270   
VIRGINIA                     116    116             116               116   
WASHINGTON                   347    347             347               347   
WEST VIRGINIA                  1      1               1                 1   
WISCONSIN                     34     34              34                34   

                      Purch Party  event name  Name of Primary Act  \
venue_state                                                          
ALABAMA                        

# Conclusion

Now looking back at this I admire the hardwork of those  that have to  do data cleaning on a daily basis. THis not only takes a lot of hard work but also a lot of patience. There were times in theis lab where I almost gave up, times where I didnt think that I coukd do it. But they key to not giving up is to take breaks in between so that you have fresh eyes each time that you are relooking at the data.

Some of the final things that I saw in both datasets were that (in the attendance dataset) the most school and states had 180 days in their school day. I found this to match up with the average amont of school days in the US. One of the interesting parts of the second dataset( the sales dataset) was when I grouped all the data from venue state and found that the city with the most sales or events was Washingon! I found this interesting becuase Washington is where I live.

# Acknowledgements

I want to aknowledge Rihan, Ms. Sconyers, and Navya for helping me with this lab.